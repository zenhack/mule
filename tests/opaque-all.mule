#XFAIL: type member
(fn x. {}) : { type t } -> {}
