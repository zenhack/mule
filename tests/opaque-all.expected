
Desugared:

	(: (fn x {})
	 ((exist $6 (record (types ((t $6))) (values ()))) ->
	  (record (types ()) (values ()))))

inferred type:

	((exist a (record (types ((t a))) (values ()))) ->
	 (record (types ()) (values ())))

Runtime term:

	(Lam (0 () (Record ())))

Evaluated:

	(Lam (0 () (Record ())))
